modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -75,6 +77,9 @@ public TypeScriptAngularClientCodegen() {
         this.cliOptions.add(new CliOption(TAGGED_UNIONS,
                 "Use discriminators to create tagged unions instead of extending interfaces.",
                 SchemaTypeUtil.BOOLEAN_TYPE).defaultValue(Boolean.FALSE.toString()));
+        this.cliOptions.add(new CliOption(PROVIDED_IN_ROOT,
```

Comment:
wouldn't you want to set the property depending on the version, i.e. enabled for angular 6.0.0+ instead of adding a new cli option? similar to the `useRxJS6` option...